src/App.tsx

```diff
@@ -1,10 +1,68 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { FilterType } from './types/filter';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
 
-const USER_ID = 0;
+const USER_ID = 10344;
+
+const FETCH_ERROR = 'Failed to fetch todos from the server. '
++ 'Please check your internet connection and try again later.';
+
+const filterActive = (
+  data: Todo[],
+) => data.filter((todo) => todo.completed === false);
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const activeLength = useMemo(
+    () => filterActive(todos).length,
+    [todos],
+  );
+
+  const completedLength = useMemo(
+    () => todos.length - activeLength,
+    [todos],
+  );
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout | undefined;
+
+    getTodos(USER_ID)
+      .then((todosFromApi) => {
+        setTodos(todosFromApi);
+        setLoading(true);
```

Comment:
the callback you write in the `then` method runs when the promise is fulfilled - i.e. when you get a response to your request. instead you should set it `true` before sending the request